### PR TITLE
B1M2-5354

### PR DESCRIPTION
**Pull request by prushin**

### CODE REVIEW

File: src/LndClient/RPC/Katip.hs [hunk 1]
```diff
@@ -174,3 +177,28 @@ closeChannelSync env mConn req =
...
         Nothing -> closeChannelRecursive mVar0 $ n - 1
+
+trackPaymentSync ::
+  (KatipContext m, MonadUnliftIO m) =>
+  LndEnv ->
+  TrackPaymentRequest ->
+  m (Either LndError Payment)
+trackPaymentSync env req = do
+  mVar <- newEmptyMVar
+  void $
```
Comment (tim2CF) [inline src/LndClient/RPC/Katip.hs:188]: Use better `withSpawnLink` combinator

File: src/LndClient/RPC/Katip.hs [hunk 2]
```diff
@@ -174,3 +177,28 @@ closeChannelSync env mConn req =
...
+        (void . tryPutMVar mVar)
+        env
+        req
+  waitTrackResult mVar 10
+  where
+    waitTrackResult _ (0 :: Int) = do
+      $(logTM) (newSev env ErrorS) "Track Payment timeout expired"
+      return $ Left $ LndError "Track Payment timeout expired"
+    waitTrackResult mVar0 n = do
+      sleep $ MicroSecondsDelay 1000000
```
Comment (tim2CF) [inline src/LndClient/RPC/Katip.hs:200]: Maybe let's use smaller sleep? 1sec is quite a lot. If first message is current status of the payment - it should arrive pretty fast, and we need only it

File: src/LndClient/RPC/Silent.hs [hunk 3]
```diff
@@ -149,3 +152,27 @@ closeChannelSync env conn req = do
       case upd of
         Just res -> return $ Right res
         Nothing -> closeChannelRecursive mVar0 (n -1)
+
+trackPaymentSync ::
```
Comment (tim2CF) [inline src/LndClient/RPC/Silent.hs:156]: Same issues like in Katip version

**Review by tim2CF — CHANGES_REQUESTED** [review]

File: src/LndClient/RPC/Katip.hs (hunk 1 above)
Comment (prushin) [reply src/LndClient/RPC/Katip.hs:188]: Okay

File: src/LndClient/RPC/Katip.hs (hunk 2 above)
Comment (prushin) [reply src/LndClient/RPC/Katip.hs]: yep, decreased 10 times

**Review by tim2CF — APPROVED** [review]
Let's wait for tests and merge if code is ok

**Pull request merged**